cdap-security/src/main/java/io/cdap/cdap/security/authorization/AccessControllerWrapper.java

```diff
@@ -0,0 +1,279 @@
+  public AuthorizationResponse createRole(Principal caller, Role role) throws AccessException {
+    try {
+      accessControllerV1.createRole(role);
+    } catch (UnauthorizedException e) {
+        return createAuthResultFromUnauthorizedExp(e);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+    return AuthorizationResponse.Builder.defaultNotRequired();
+  }
+
+  @Override
+  public AuthorizationResponse dropRole(Principal caller, Role role) throws AccessException {
+    try {
+      accessControllerV1.dropRole(role);
+    } catch (UnauthorizedException e) {
+      return createAuthResultFromUnauthorizedExp(e);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+    return AuthorizationResponse.Builder.defaultNotRequired();
+  }
+
+  @Override
+  public AuthorizationResponse addRoleToPrincipal(Principal caller, Role role, Principal principal)
+    throws AccessException {
+    try {
+      accessControllerV1.addRoleToPrincipal(role, principal);
+    } catch (UnauthorizedException e) {
+      return createAuthResultFromUnauthorizedExp(e);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+    return AuthorizationResponse.Builder.defaultNotRequired();
+  }
+
+  @Override
+  public AuthorizationResponse removeRoleFromPrincipal(Principal caller, Role role, Principal principal)
+    throws AccessException {
+    try {
+      accessControllerV1.removeRoleFromPrincipal(role, principal);
+    } catch (UnauthorizedException e) {
+      return createAuthResultFromUnauthorizedExp(e);
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+    return AuthorizationResponse.Builder.defaultNotRequired();
+  }
+
+  @Override
+  public AuthorizationResult<Set<Role>> listRoles(Principal caller, Principal principal) throws AccessException {
+    Set<Role> roleSet;
+    try {
+      roleSet = accessControllerV1.listRoles(principal);
+    } catch (UnauthorizedException e) {
+      //Incase of UnauthorizedException , the set will be empty
+      return new AuthorizationResult<>(Collections.emptySet(), AuthorizationResponse.Builder.defaultNotRequired());
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+
+    return new AuthorizationResult<>(roleSet, AuthorizationResponse.Builder.defaultNotRequired());
+  }
+
+  @Override
+  public AuthorizationResult<Set<Role>> listAllRoles(Principal caller) throws AccessException {
+    Set<Role> roleSet;
+    try {
+      roleSet = accessControllerV1.listAllRoles();
+    } catch (UnauthorizedException e) {
+      return new AuthorizationResult<>(Collections.emptySet(), AuthorizationResponse.Builder.defaultNotRequired());
+    } catch (Exception e) {
+      throw AuthEnforceUtil.propagateAccessException(e);
+    }
+    return new AuthorizationResult<>(roleSet, AuthorizationResponse.Builder.defaultNotRequired());
```

Comment:
nit: you don't really need to declare variable and then return outside of the try/catch. You can directly return within `try`